dask/dataframe/io/tests/test_parquet.py

```diff
@@ -2518,3 +2519,28 @@ def _arrow_table_to_pandas(
 
     # this should not fail, but instead produce timestamps that are in the valid range
     dd.read_parquet(str(tmpdir), engine=ArrowEngineWithTimestampClamp).compute()
+
+
+@write_read_engines_xfail
+def test_partitioned_preserve_index(tmpdir, write_engine, read_engine):
+
+    if write_engine == "pyarrow" and pa.__version__ < LooseVersion("0.15.0"):
+        pytest.skip("PyArrow>=0.15 Required.")
+
+    tmp = str(tmpdir)
+    size = 1_000
+    npartitions = 4
+    b = np.arange(npartitions).repeat(size // npartitions)
+    data = pd.DataFrame(
+        {
+            "myindex": np.arange(size),
+            "A": np.random.random(size=size),
+            "B": pd.Categorical(b),
+        }
+    ).set_index("myindex")
+    df1 = dd.from_pandas(data, npartitions=npartitions)
+    df1.to_parquet(tmp, partition_on="B", engine=write_engine)
+
+    expect = df1[df1["B"] == 1]
+    got = dd.read_parquet(tmp, engine=read_engine, filters=[("B", "==", 1)])
+    assert_eq(expect, got, check_divisions=False)
```

Comment:
Do you prefer comparing against a concrete result?

```suggestion
    expect = data[data["B"] == 1]
    got = dd.read_parquet(tmp, engine=read_engine, filters=[("B", "==", 1)])
    assert_eq(expect, got)
```